testsuite/MDAnalysisTests/coordinates/test_chemfiles.py

```diff
@@ -0,0 +1,152 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import absolute_import
+
+import pytest
+
+import MDAnalysis as mda
+import numpy as np
+from unittest import TestCase
+
+from MDAnalysis.coordinates.chemfiles import ChemfilesReader, ChemfilesWriter
+
+from MDAnalysisTests import datafiles, tempdir
+from MDAnalysisTests.coordinates.base import (
+    MultiframeReaderTest, BaseWriterTest, BaseReference
+)
+
+
+class ChemfilesXYZReference(BaseReference):
+    def __init__(self):
+        super(ChemfilesXYZReference, self).__init__()
+        self.trajectory = datafiles.COORDINATES_XYZ
+        self.topology = datafiles.COORDINATES_XYZ
+        self.reader = ChemfilesReader
+        self.writer = ChemfilesWriter
+        self.ext = 'xyz'
+        self.volume = 0
+        self.dimensions = np.zeros(6)
+        self.dimensions[3:] = 90.0
+
+
+class TestChemfilesReader(MultiframeReaderTest):
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return ChemfilesXYZReference()
+
+
+class TestChemfilesWriter(BaseWriterTest):
+    @staticmethod
+    @pytest.fixture()
+    def ref():
+        return ChemfilesXYZReference()
+
+    # Disable 'test_no_container' as it try to open a file for writing without
+    # extension.
+    def test_no_container(self, ref):
+        pass
```

Comment:
what is the exact issue here? Could we throw an appropriate exception when to file extension is given and check in a test if the exception is thrown?

Reply:
This test try to open a file named 'foo' to write to it and then read it back

https://github.com/MDAnalysis/mdanalysis/blob/abc90493f27ad160e311405f78d4377ac7a357ef/testsuite/MDAnalysisTests/coordinates/base.py#L594-L600

this fails with an exception `chemfiles.misc.ChemfilesError: file at 'foo' does not have an extension, provide a format name to read it`.

Do you mean I should add a test in `test_chemfiles.py` for this behavior?

Reply:
yeah adding an explicit test for this would be nice as the behavior is fundamentally different from other writers. Would the following work?

```python
def test_no_container_raises(self, ref):
    with pytest.raises(TypeError): 
        ref.writer('foo') 
```

Reply:
Should the ChemfilesError be converted into TypeError?

Reply:
Oh I would keep the chemfilesError